### PR TITLE
Add EventListener support #339

### PR DESCRIPTION
**Pull request by brbrr**
https://github.com/appium/ruby_lib/issues/339

### CODE REVIEW

File: lib/appium_lib/driver.rb [hunk 1]
```diff
@@ -337,6 +339,9 @@ def initialize(opts = {})
       # !!'constant' == true
       @appium_debug = appium_lib_opts.fetch :debug, !!defined?(Pry)
 
+      # to pass it in Selenium.new
+      @listener = opts[:listener]
```
Comment (bootstraponline) [inline lib/appium_lib/driver.rb]: This shouldn't be a top level opt.

`appium_lib_opts[:listener]`

File: lib/appium_lib/driver.rb [hunk 2]
```diff
@@ -476,7 +481,8 @@ def start_driver
 
       begin
         driver_quit
-        @driver = Selenium::WebDriver.for :remote, http_client: @client, desired_capabilities: @caps, url: server_url
+        @driver = Selenium::WebDriver.for :remote, http_client: @client,
+                                                   desired_capabilities: @caps, url: server_url, listener: @listener
```
Comment (bootstraponline) [inline lib/appium_lib/driver.rb]: does this work when `@listener` is nil?

**Comment by bootstraponline** [comment]
This feature should have a test to verify it works and doesn't regress in the future.

**Comment by KazuCocoa** [comment]
I merged in Add EventListener to Driver #389

**Pull request closed without merging**